sphinx_immaterial/nav_adapt.py

```diff
@@ -258,6 +272,8 @@ class ObjectIconInfo(NamedTuple):
     ("cpp", "var"): ObjectIconInfo(icon_class="alias", icon_text="V"),
     ("cpp", "type"): ObjectIconInfo(icon_class="alias", icon_text="T"),
     ("cpp", "namespace"): ObjectIconInfo(icon_class="alias", icon_text="N"),
+    ("cpp", "templateParam"): ObjectIconInfo(icon_class="data", icon_text="P"),
```

Comment:
I recently wondered if we could wrap the template param names in angle brackets for the secondary toc.
![image](https://user-images.githubusercontent.com/14963867/162333065-b2ea5190-6b68-4830-8d56-8275228ed478.png)
Could be less garbled like "\<T A B C N>std::array". Screenshot is from my more-extensions branch's changes in which I added a bunch of template params to see if they could be wrapped into multiple lines in lengthy conditions.

Reply:
For the tensorstore docs I am using a custom C++ autosummary extension (kind of like the custom Python autosummary extension I'm using), and then I manually set a `toc_title` attribute on all of the signatures, which then gets picked up by `object_toc` so this default name generation doesn't apply.

The reason the template parameter names are included is that the current logic is to concatenate all of the desc_name elements, but Sphinx generates the template parameter names as desc_name as well.  I have a fix for that (which would just exclude the template parameter names).

If we wanted to include them, we should presumably do it in the normal way, e.g. `std::array<T, A, B, C, N>`.  Not sure if that is desirable, though --- may make the TOC entry too long in some cases; it might be better to only do that for template specializations.

Reply:
I'm ok with excluding them for template declarations. I also like the idea of appending the template names for template specializations.

> For the tensorstore docs I am using a custom C++ autosummary extension 

Was breathe not a viable option due to the babel build system or something? (just curious)

Reply:
I did look into breathe --- in fact when I first began creating the TensorStore documentation I selected sphinx with the intention of using breathe.

However, there were a number of things I wanted that breathe didn't really help with:

- generating a custom autosummary similar to my python autosummary
- custom transformations of declarations, to hide internal types, convert enable_if uses to c++20 requires clauses, convert certain patterns to c++20 explicit(bool).

Additionally, doxygen doesn't parse c++ perfectly, and it has its own markup --- I want to just use rST in my doc comments without having to put \rst and \endrst.

Plus doxygen is an additional dependency to have to deal with.

I found that there is a python libclang package, in pypi that is kept up to date and statically links clang, and using that it is pretty easy to extract declarations and comments from c++ source code, and it is guaranteed to parse correctly since the code is tested using clang.

Using that plus the c++ parser included in the sphinx c++ domain I have been able to accomplish what I want.